agent/csi/mod_csi_docker.go

```diff
@@ -498,26 +512,40 @@ func (d *dockerBridge) Create(req *dvol.CreateRequest) error {
 
 func (d *dockerBridge) List() (*dvol.ListResponse, error) {
 
-	// Create a new gRPC, CSI client.
-	c, err := d.cs.dial(d.ctx)
-	if err != nil {
-		d.ctx.Errorf("docker-csi-bridge: List: client failure: %v", err)
-		return nil, err
-	}
-	defer c.Close()
+	var (
+		vols []*csi.VolumeInfo
+	)
 
-	// Create a new CSI Controller client.
-	cc := csi.NewControllerClient(c)
+	// If the service is CSI-NFS then list is handled differently.
+	if d.isCSINFS() {
```

Comment:
For `ListVolumes`, we weren't returning the cached volumes for NFS.  But somehow I've still done this wrong, as the results I get are inconsistent.  If I have two volumes, I know the cache gets warmed correctly:
```
DEBU[0000] getNFSVolMap: cached prepopulated nfs volume  nfsExport=/data nfsHost=192.168.75.2 nfsVolName=test time=1507264815294
DEBU[0000] getNFSVolMap: cached prepopulated nfs volume  nfsExport=/data nfsHost=192.168.75.2 nfsVolName=test3 time=1507264815296
```

But when I do `docker volume ls`, I get inconsistent results, and the volumes always have the same name:
```
[root@localhost vagrant]# docker volume ls
DRIVER              VOLUME NAME
rexray              test
rexray              test
[root@localhost vagrant]# docker volume ls
DRIVER              VOLUME NAME
rexray              test3
rexray              test3
```

I'm not sure what I did wrong there. I was trying to get just get all the volumes out of the cache, and let the existing logic pull the name out of the metadata.